lms/djangoapps/verify_student/tasks.py

```diff
@@ -40,3 +90,48 @@ def send_verification_status_email(context):
         )
     except SMTPException:
         log.warning(u"Failure in sending verification status e-mail to %s", dest_addr)
+
+
+@task(
+    base=BaseSoftwareSecureTask,
+    bind=True,
+    default_retry_delay=settings.SOFTWARE_SECURE_REQUEST_RETRY_DELAY,
+    max_retries=settings.SOFTWARE_SECURE_RETRY_MAX_ATTEMPTS,
+    routing_key=SOFTWARE_SECURE_VERIFICATION_ROUTING_KEY,
+)
+def send_request_to_ss_for_user(self, user_verification_id, copy_id_photo_from):
+    """
+    Assembles a submission to Software Secure.
+
+    Keyword Arguments:
+        user_verification_id (int) SoftwareSecurePhotoVerification model object identifier.
+        copy_id_photo_from (SoftwareSecurePhotoVerification): If provided, re-send the ID photo
+                data from this attempt.  This is used for re-verification, in which new face photos
+                are sent with previously-submitted ID photos.
+    Returns:
+        request.Response
+    """
+    log.info('=>New Verification Task Received')  # todo -- remove before merge.
+    user_verification = SoftwareSecurePhotoVerification.objects.get(id=user_verification_id)
+    try:
+        headers, body = user_verification.create_request(copy_id_photo_from)
+        response = requests.post(
+            settings.VERIFY_STUDENT["SOFTWARE_SECURE"]["API_URL"],
+            headers=headers,
+            data=simplejson.dumps(body, indent=2, sort_keys=True, ensure_ascii=False).encode('utf-8'),
+            verify=False
+        )
+        return response
+    except Exception as exc:  # pylint: disable=bare-except
+        log.error(
+            (
+                'Retrying sending request to Software Secure for user: %r, Receipt ID: %r '
+                'attempt#: %s of %s'
+            ),
+            user_verification.user.username,
+            user_verification.receipt_id,
+            self.request.retries,
+            settings.SOFTWARE_SECURE_RETRY_MAX_ATTEMPTS,
+        )
+        log.error(str(exc))
+        raise self.retry()
```

Comment:
Why is the raise necessary? The retry works without it ([Ref](https://coderbook.com/@marcus/how-to-automatically-retry-failed-tasks-with-celery/))

Reply:
Ok.